### PR TITLE
feat(biome_grit_formatter): initial infrastructure

### PR DESCRIPTION
**Pull request by abidjappie**
## Summary

Set up the basic infrastructure required for further work on #2476 

### Implementation

- Run `just new-crate biome_grit_formatter`
- Update `Cargo.toml` with relevant settings
- Update `xtask/codegen/src/formatter`
- Run `just gen-formatter`

## Test Plan

Cross comparison with other Biome formatters.

### CODE REVIEW

**Review by abidjappie — COMMENTED** [review]
The code generator created `src/js`, do we need to keep this or can it be removed?

**Comment by ematipico** [comment]
> The code generator created `src/js`, do we need to keep this or can it be removed?

This isn't correct. You have to update the codegen here:

https://github.com/biomejs/biome/blob/7095ed0c143212c9f649d3b6b326650ae0416d1d/xtask/codegen/src/formatter.rs#L785-L786

**Comment by ematipico** [comment]
It's possible you'll have to update this too:

https://github.com/biomejs/biome/blob/7095ed0c143212c9f649d3b6b326650ae0416d1d/xtask/codegen/src/formatter.rs#L553-L559

**Comment by abidjappie** [comment]
@ematipico Thank you for the feedback, as always!

I think I understand it a bit better now, although I am still a bit unclear on how the GritQL nodes are best categorized. I'll re-iterate again soon.

**Comment by abidjappie** [comment]
I'm not sure if covering all of these should be in the scope of this PR.

>  ### Configuration
> We need to add GritQL as a language in the configuration format, and let it support the common options around indentation, line length and type of newlines.
 > - Add test suite
 > - Introduce a can_format_grit_yet flag into biome_service
 > - Comment placement
 > - Handle biome ignore: format suppression comments in appropriate places
 
 Additionally, I noticed that https://github.com/biomejs/biome/pull/1207 also added some additions like:
 - Readme
 - Git related
 - More...

File: xtask/codegen/src/formatter.rs [hunk 1]
```diff
@@ -782,8 +792,15 @@ fn get_node_concept(
             // TODO: implement formatter
             LanguageKind::Graphql => NodeConcept::Auxiliary,
 
-            // TODO: I will handle formatting in a follow-up PR.
-            LanguageKind::Grit => NodeConcept::Auxiliary,
+            LanguageKind::Grit => match name {
+                _ if name.contains("Operation") || name.contains("Pattern") => NodeConcept::Pattern,
+                _ if name.contains("Predicate") => NodeConcept::Predicate,
+                _ if name.ends_with("Definition") => NodeConcept::Declaration,
+                _ if matches!(name, "CodeSnippet") || name.ends_with("Literal") => {
```
Comment (arendjr) [inline xtask/codegen/src/formatter.rs:799]: I would've used `name == "CodeSnippet"` here ;)

**Review by arendjr — APPROVED** [review]
Great start!

**Comment by arendjr** [comment]
> I'm not sure if covering all of these should be in the scope of this PR

No, I think I'll just merge as is, and you can add options like this in follow-up PRs. That will allow us to iterate a bit faster, and you'll have to worry less about merge issues.

**Comment by arendjr** [comment]
There just seems to be an issue with an unused dependency still.

File: crates/biome_grit_formatter/Cargo.toml [hunk 2]
```diff
@@ -13,21 +13,13 @@ version              = "0.0.0"
...
-biome_formatter_test = { path = "../biome_formatter_test" }
-biome_grit_parser    = { path = "../biome_grit_parser" }
-biome_parser         = { path = "../biome_parser" }
-biome_service        = { path = "../biome_service" }
-countme              = { workspace = true, features = ["enable"] }
-serde                = { workspace = true, features = ["derive"] }
-serde_json           = { workspace = true }
-tests_macros         = { path = "../tests_macros" }
+serde      = { workspace = true, features = ["derive"] }
+serde_json = { workspace = true }
```
Comment (abidjappie) [inline crates/biome_grit_formatter/Cargo.toml:22]: I cleared all the unused dependencies (those that weren't imported) except for `serde`. I hope `udeps` does not complain again.

**Pull request merged**